indy-besu/docs/design/auth.md

```diff
@@ -0,0 +1,187 @@
+# Auth model
+
+## Roles
+
+| Label    | Value     |
+|----------|-----------|
+| Trustee  | 1         |
+| Endorser | 2         |
+| Steward  | 3         |
+
+## Account role management
+
+| Contract    | Method     | Value      | Required Role | Action Description                       |
+|-------------|------------|------------|---------------|------------------------------------------|
+| RoleControl | hasRole    | -          | any           | Check if an account has a requested role |
+| RoleControl | getRole    | -          | any           | Get account role                         |
+| RoleControl | assignRole | Trustee    | Trustee       | Assign Trustee role to an account        |
+| RoleControl | assignRole | Endorser   | Trustee       | Assign Endorser role to an account       |
+| RoleControl | assignRole | Steward    | Trustee       | Assign Steward role to an account        |
+| RoleControl | revokeRole | Trustee    | Trustee       | Revoke Trustee role from an account      |
+| RoleControl | revokeRole | Endorser   | Trustee       | Assign Endorser role to an account       |
+| RoleControl | revokeRole | Steward    | Trustee       | Assign Steward role to an account        |
+
+## Validator nodes management
+
+| Contract         | Method          | Required Role | Action Description                      |
+|------------------|-----------------|---------------|-----------------------------------------|
+| ValidatorControl | getValidators   | any           | Get the list of current validator nodes |
+| ValidatorControl | addValidator    | Steward       | Add new validator node                  |
+| ValidatorControl | removeValidator | Steward       | Remove validator node                   |
+
+## DID Document management
+
+| Contract      | Method                         | Required Role               | Action Description              |
+|---------------|--------------------------------|-----------------------------|---------------------------------|
+| DidRegistry   | createDid                      | Trustee, Endorser, Steward  | Create a new DID Document       |
+| DidRegistry   | updateDid                      | DID owner                   | Update DID an existing Document |
+| DidRegistry   | deactivateDid                  | DID owner                   | Deactivate an existing DID      |
+| DidRegistry   | resolveCredentialDefinition    | any                         | Resolve DID Document for a DID  |
```

Comment:
I think this is probably resolveDid?

Reply:
Thank you for pointing out the typo. Yes, it should be ‘resolveDid’. I’ve updated the documentation.